src/ugeneui/src/Main.cpp

```diff
@@ -496,42 +496,38 @@ int main(int argc, char **argv) {
         userAppSettings->setFileStorageDir(FileAndDirectoryUtils::getAbsolutePath(cmdLineRegistry->getParameterValue(CMDLineCoreOptions::FILE_STORAGE_DIR)));
     }
 
-    bool trOK = false;
+    // Set translations if needed: use value in the settings or environment variables to override.
+    // The default case 'en' does not need any files: the values for this locale are hardcoded in the code.
     QTranslator translator;
+    QStringList traceLogFromTranslator;    // Details about translator initialization to log when the log system is initialized.
+
+    // The file specified by user has the highest priority in the translations lookup order.
     QStringList envList = QProcess::systemEnvironment();
-    QString envTranslation = findKey(envList, "UGENE_TRANSLATION");
-    if (!envTranslation.isEmpty()) {
-        trOK = translator.load(QString("transl_") + envTranslation, AppContext::getWorkingDirectoryPath());
-        settings->setValue("UGENE_CURR_TRANSL", envTranslation);
-    }
     QString envTranslationFile = findKey(envList, "UGENE_TRANSLATION_FILE");
-    if (!envTranslationFile.isEmpty()) {
-        trOK = translator.load(envTranslationFile);
-        settings->setValue("UGENE_CURR_TRANSL", QFileInfo(envTranslationFile).fileName().right(2));
-    }
-
-    if (!trOK) {
-        // set translations
-        QString transFile[] = {
+    if (envTranslationFile.isEmpty() || !translator.load(envTranslationFile)) {
+        if (!envTranslationFile.isEmpty()) {
+            traceLogFromTranslator << "Failed to load translation file: " + envTranslationFile;
+        }
+        QStringList translationFileList = {
+            "transl_" + findKey(envList, "UGENE_TRANSLATION"),
             userAppSettings->getTranslationFile(),
-            "transl_en"};
-        for (int i = transFile[0].isEmpty() ? 1 : 0; i < 3; ++i) {
-            if (!translator.load(transFile[i], AppContext::getWorkingDirectoryPath())) {
-                fprintf(stderr, "Translation not found: %s\n", transFile[i].toLatin1().constData());
-            } else {
-                settings->setValue("UGENE_CURR_TRANSL", transFile[i].right(2));
-                trOK = true;
+            "transl_" + QLocale::system().name().left(2).toLower()};
+        // Keep only valid entries.
+        translationFileList.removeAll("");
+        translationFileList.removeAll("transl_");
+        translationFileList.removeDuplicates();
+        // Use the first translation from the list that works.
+        for (const QString &translationFile : qAsConst(translationFileList)) {
+            if (translationFile == "transl_en" || translator.load(translationFile, AppContext::getWorkingDirectoryPath())) {
                 break;
             }
-        }
-        if (!trOK) {
-            fprintf(stderr, "No translations found, exiting\n");
-            return 1;
+            traceLogFromTranslator << "Translation not found: " + translationFile;
```

Comment:
The same, better to translate it if the target language will bee different from English or Russian (who knows, maybe in the future we will support other languages)